saclock: add "monochrome" setting for Bangle.js 2

rigrig opened the pull request:
Add "monochrome" setting to Simple Analog Clock   

Before it always was monochrome, disabling the setting makes it use multiple theme colors.

Only for Bangle.js 2, because part of the "simple" is that we draw to a graphics buffer, and B1 doesn't have enough memory for a colour buffer. (Drawing directly would involve sprinkling y-offsets and "redraw background" all over the code; doable, but no longer simple)

Preview: https://bangle.tubul.net/?id=saclock

thyttan commented:
Thanks!

I get some weirdness with graphics, depending on settings.

![image](https://github.com/espruino/BangleApps/assets/97237430/cdcd587e-4ae5-4e89-88e3-f8d84a10e98c)
_System Light theme. App settings hide widgets. Monochrome on. White lines at right and bottom border._

![image](https://github.com/espruino/BangleApps/assets/97237430/7252a3e6-dbec-41e1-822c-818dfa2a1bb5)
_System Dark theme. App settings hide widgets. Monochrome on. Looks a little bit like the circle doesn't fit on screen? But maybe the circle would look like similar even if a little smaller._

![image](https://github.com/espruino/BangleApps/assets/97237430/911416d6-d3c6-4956-9c69-f6c65f1f9efd)
_System light theme. App settings show widgets. Monochrome on._

![image](https://github.com/espruino/BangleApps/assets/97237430/6d5b981b-d9f3-4f9a-87dc-ae25017d2615)
_System light theme. App settings show widgets. Monochrome on._

I didn't go through all setting combinations.

rigrig commented:
Whoops, I was planning to look at this much earlier, but so far only got to the "confirm it's broken" bit (for about every theme except the ones I tested it on initially :-()   
Thanks for looking at it, @thyttan   
(I expect I'll have time to fix it this weekend.)

rigrig commented on apps/saclock/app.js:
```diff
@@ -67,13 +71,18 @@ let clock = new (require("ClockFace"))({
     drawHand(ml, mr); // minute hand
     buf.fillCircle(c, c, Math.floor(hw)); // hands joiner
+    if (!this.multicol && !g.theme.dark) {
+        // apparently 1-bit g.drawImage swaps fg/bg colors for light themes?
```
I'm not sure why we need this to prevent inverted colors for monochrome on light theme. But it works now...

rigrig commented:
Fixed the drawing offsets, and made the outer rim 1px smaller, so now the circle definitely fits.   
Also added some more screenshots.
And unbroke the settings (wow, the first version of this PR was *really* broken)

thyttan commented on apps/saclock/screenshot3.png:
How is this particular colour combination achieved?

rigrig commented:
I'm using a custom theme:

    {"fg":2047,"bg":0,"fg2":63488,"bg2":0,"fgH":0,"bgH":65504,"dark":true,"name":"Colorful"}

So I guess it's better to leave out that screenshot (and maybe add some for the default themes) instead,
to prevent confusion.

October 9, 2023 at 1:22 AM, "thyttan" ***@***.***> wrote:
> 
> ***@***.***** commented on this pull request.
> 
> ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
> 
> On apps/saclock/screenshot3.png https://github.com/espruino/BangleApps/pull/3024#discussion_r1349794162 :
> 
> How is this particular colour combination achieved?
> 
> —
> Reply to this email directly, view it on GitHub https://github.com/espruino/BangleApps/pull/3024#pullrequestreview-1663618258 , or unsubscribe https://github.com/notifications/unsubscribe-auth/AAWOD7DVCBJETSOULXZHD23X6MYTTAVCNFSM6AAAAAA5DRFNNSVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTMNRTGYYTQMRVHA .
> You are receiving this because you authored the thread.
>

--
No one remembers the singer.

thyttan commented:
> I'm using a custom theme:
> 
>     {"fg":2047,"bg":0,"fg2":63488,"bg2":0,"fgH":0,"bgH":65504,"dark":true,"name":"Colorful"}
> 
> So I guess it's better to leave out that screenshot (and maybe add some for the default themes) instead,
> to prevent confusion.
> 
> October 9, 2023 at 1:22 AM, "thyttan" ***@***.***> wrote:
> > 
> > ***@***.***** commented on this pull request.
> > 
> > ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
> > 
> > On apps/saclock/screenshot3.png https://github.com/espruino/BangleApps/pull/3024#discussion_r1349794162 :
> > 
> > How is this particular colour combination achieved?
> > 
> > —
> > Reply to this email directly, view it on GitHub https://github.com/espruino/BangleApps/pull/3024#pullrequestreview-1663618258 , or unsubscribe https://github.com/notifications/unsubscribe-auth/AAWOD7DVCBJETSOULXZHD23X6MYTTAVCNFSM6AAAAAA5DRFNNSVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTMNRTGYYTQMRVHA .
> > You are receiving this because you authored the thread.
> >
> 
> --
> No one remembers the singer.
> 

Yes I worked it out and is currently doing the same! 😃 Maybe in a future PR the user could get the option to set colours directly?

No need to drop that screenshot in my view 🙂

gfwilliams commented:
Looks good - thanks!
